### DISCUSSION TITLE
Import pagefind via shared CDN

### DISCUSSION THREAD

**Discussion by JosefJezek** [withastro/starlight#1858] [Feature Requests]
### What version of `starlight` are you using?

0.22.2

### What is your idea?

Import pagefind pkg via shared CDN, ex. [esm.sh](https://esm.sh/)

The pagefind pkg is here:

```js
import '@pagefind/default-ui/css/ui.css';
const { PagefindUI } = await import('@pagefind/default-ui');
```
https://github.com/withastro/starlight/blob/main/packages/starlight/components/Search.astro

### Why is this feature necessary?

- Shared CDN packages between apps reduce requests to servers and can help our planet.
  - https://stackoverflow.com/questions/62119426/is-can-a-resource-js-file-from-a-cdn-be-shared-across-websites-browser-tabs
- Build tool doesn't need to re-build dependencies, which conserves energy and reduces carbon emissions.
- more info https://github.com/withastro/roadmap/discussions/919

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Would need testing, but I’m fairly sure those StackOverflow answers you linked are outdated — in general cross-origin caching was deprecated for security reasons (to prevent cached assets being used to work around newer cross-site tracking limitations).

So switching to a third-party CDN in this case would likely damage performance: it would be slower to establish the connection to a second server instead of reusing the open connection to the site’s origin server.

**Reply by JosefJezek** [reply]
ok, thank you... we have HTTP/2 for data streaming, new request is not needed
